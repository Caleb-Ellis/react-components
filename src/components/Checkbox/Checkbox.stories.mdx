import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs/blocks";
import { useState } from "react";
import Checkbox from "./Checkbox";

<Meta
  title="Checkbox"
  component={Checkbox}
  argTypes={{
    label: {
      control: {
        type: "text",
      },
      defaultValue: "Checkbox",
    },
    disabled: {
      control: {
        type: "boolean",
      },
    },
    required: {
      control: {
        type: "boolean",
      },
    },
    indeterminate: {
      control: {
        type: "boolean",
      },
    },
  }}
/>

export const Template = (args) => <Checkbox {...args} />;

### Checkbox

TODO: add description.

### Props

<ArgsTable of={Checkbox} />{" "}

### Default

<Canvas>
  <Story name="Default">{Template.bind({})}</Story>
</Canvas>

### Disabled

<Canvas>
  <Story name="Disabled">
    <Checkbox label="Disabled" disabled></Checkbox>
  </Story>
</Canvas>

### Required

<Canvas>
  <Story name="Required">
    <Checkbox label="Required" required></Checkbox>
  </Story>
</Canvas>

### Indeterminate

<Canvas>
  <Story name="Indeterminate">
    <Checkbox label="Indeterminate" indeterminate></Checkbox>
  </Story>
</Canvas>
