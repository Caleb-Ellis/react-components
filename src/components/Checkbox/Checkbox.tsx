import PropTypes from "prop-types";
import React, { HTMLProps } from "react";

import CheckableInput from "../CheckableInput";

type Props = {
  label: string;
} & HTMLProps<HTMLElement>;

const Checkbox = ({ type, label, ...checkboxProps }: Props): JSX.Element => {
  return (
    <CheckableInput
      type="checkbox"
      label={label}
      {...checkboxProps}
    ></CheckableInput>
  );
};

Checkbox.propTypes = {
  label: PropTypes.string.isRequired,
};

export default Checkbox;
