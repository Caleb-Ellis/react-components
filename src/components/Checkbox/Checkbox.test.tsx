import { shallow } from "enzyme";
import React from "react";

import Checkbox from "./Checkbox";

describe("Checkbox ", () => {
  it("renders a checkbox", () => {
    const wrapper = shallow(<Checkbox label="Test checkbox"></Checkbox>);
    expect(wrapper).toMatchSnapshot();
  });
});
