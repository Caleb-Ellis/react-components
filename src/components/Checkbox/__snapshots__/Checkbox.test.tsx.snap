// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox  renders 1`] = `
<CheckableInput
  label="Test checkbox"
  type="checkbox"
/>
`;
